SConstruct

```diff
@@ -652,11 +652,24 @@ if selected_platform in platform_list:
     # Configure compiler warnings
     if env.msvc:  # MSVC
         # Truncations, narrowing conversions, signed/unsigned comparisons...
-        disable_nonessential_warnings = ["/wd4267", "/wd4244", "/wd4305", "/wd4018", "/wd4800"]
+        disable_nonessential_warnings = [
+            "/wd4018",
+            "/wd4100",
+            "/wd4127",
+            "/wd4201",
+            "/wd4244",
+            "/wd4245",
+            "/wd4267",
+            "/wd4305",
+            "/wd4514",
+            "/wd4714",
+            "/wd4800",
+            "/wd4820",
+        ]
```

Comment:
In #66595 I refactored this a bit and added proper justifications for the warnings we skip, would be good to do the same here (or I can edit #66595 to remove the CI change and merge it first).

Reply:
Rebased #66595, I guess we can merge that one first and you can then rebase and add descriptions for the other ones.
Note that I'm also removing `/wd4018` and `/wd4800` which are no longer emitted by the current code (at least with VS 2019).

Reply:
I have changed it to use #66595 style (`/W3` for all `/W4` for extra), also removed 4018 / 4800 and changed CI to use `warnings=extra werror=yes`.